apps/contributions/views/orgs.py

```diff
@@ -129,6 +129,33 @@ def get_queryset(self):
         logger.warning("Encountered unexpected user %s", user.id)
         raise ApiConfigurationError
 
+    def destroy(self, request, pk: int) -> Response:
+        """Cancel a contribution.
+
+        This parallels the handle_delete() method on PortalContributorsViewSet.
+        It doesn't actually delete the object or update it directly, only
+        cancels the Stripe subscription. Webhook event listeners will update the
+        contribution status when it succeeds.
+        """
+        # Use get_queryset() to enforce the same permissions here as on
+        # retrieving contributions.
+        contribution = self.get_queryset().filter(id=pk).first()
+        if not contribution:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Should just be able to do

```suggestion
        contribution =  self.get_object() 
```

`.get_object` will only have access to queryset returned by `get_queryset`, which means it will raise 404 if contribution not in queryset (regardless of reason be it that contribution doesn't exist or that it exists but is not permitted for this user).